sdk/src/connections/target/adsd3500_sensor.cpp

```diff
@@ -303,6 +297,69 @@ aditof::Status Adsd3500Sensor::open() {
                          << " error: " << strerror(errno);
             return Status::GENERIC_ERROR;
         }
+
+        //Check chip status and reset if there are any errors.
+        if (m_firstRun) {
+            aditof::Status chipStatus = aditof::Status::GENERIC_ERROR;
+            uint16_t chipID;
+            uint8_t dealiasCheck[32] = {0};
+            dealiasCheck[0] = 1;
+
+            while (chipStatus != aditof::Status::OK) {
```

Comment:
I wouldn't loop forever if reading chip id continues to fail. It can try 10 times for example. Sometimes there might be other issues with the hardware, preventing the chip id to be read. And in that case we don't want any of our programs to hang.

Reply:
And if reading chip id & dealias parameters fails after N attempts, then you could log (as error) that "Cannot read chip ID " or "Cannot read dealias parameters" depending on the situation. Then you could say: "Latest ADSD3500 programming might not have been successful"

Reply:
Updated.